### PR TITLE
Fix bugs and remove unnecessary nullchecks.

### PR DESCRIPTION
**Pull request by gamerforEA**
Bugs detected by FindBugs.

### CODE REVIEW

File: common/buildcraft/builders/BlockMarker.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public boolean onBlockActivated(World world, int i, int j, int k, EntityPlayer e
 		}
 
 		TileEntity tile = world.getTileEntity(i, j, k);
-		if (tile instanceof TileMarker && !(tile instanceof TileConstructionMarker)) {
```
Comment (AEnterprise) [inline common/buildcraft/builders/BlockMarker.java]: why whas this removed?

**Comment by AEnterprise** [comment]
you are removing alot of nullchecks, they might not always seem needed but most of them are there for a reason, you can't always assume that you are getting what you asked and that the vars are filled in correctly, it doesn't hurt to have a null check to much but it will hurt if one is missing

File: common/buildcraft/builders/BlockMarker.java (hunk 1 above)
Comment (gamerforEA) [inline common/buildcraft/builders/BlockMarker.java]: "tile instanceof TileConstructionMarker" always return false.

**Comment by asiekierka** [comment]
Actually, the checks removed are very few and all redundant.

File: common/buildcraft/builders/TileQuarry.java [hunk 2]
```diff
@@ -205,7 +205,7 @@ public void updateEntity() {
 			// In this case, since idling() does it anyway, we should return.
 			return;
 		} else if (stage == Stage.MOVING) {
-			int energyUsed = this.getBattery().useEnergy(20, (int) Math.ceil(20 + getBattery().getEnergyStored() / 10), false);
+			int energyUsed = this.getBattery().useEnergy(20, (int) Math.ceil(20D + (double) getBattery().getEnergyStored() / 10), false);
```
Comment (psxlover) [inline common/buildcraft/builders/TileQuarry.java]: This should probably be:
```
int energyUsed = this.getBattery().useEnergy(20, (int) Math.ceil(20D + getBattery().getEnergyStored() / 10.0), false);
```

Comment (gamerforEA) [inline common/buildcraft/builders/TileQuarry.java]: Bytecode is identical, but your code is more compact.

Comment (psxlover) [inline common/buildcraft/builders/TileQuarry.java]: Are you sure? I don't know where Java applies the casting, so I would change the number to make sure it does a double division (10D or 10.0) or use parentesis around the casting, but if the casting applies before the division I guess it's fine like this.

Comment (gamerforEA) [inline common/buildcraft/builders/TileQuarry.java]: Casting to double, 10.0 and 10D in bytecode is identical. I check it in BytecodeOutline.

Comment (asiekierka) [inline common/buildcraft/builders/TileQuarry.java]: @psxlover - if either of the values is a double it does a double division. The Java compiler isn't that stupi.

Comment (psxlover) [inline common/buildcraft/builders/TileQuarry.java]: > @psxlover - if either of the values is a double it does a double division. The Java compiler isn't that stupi.

Yeah, I wasn't talking about that, I just wasn't sure whether the casting applied before the division or after.

**Pull request merged**